interface-definitions/service_dns_forwarding.xml.in

```diff
@@ -694,6 +694,54 @@
                   <valueless/>
                 </properties>
               </leafNode>
+              <leafNode name="dont-throttle-netmasks">
+                <properties>
+                  <help>IP address or subnet</help>
+                  <valueHelp>
+                    <format>ipv4</format>
+                    <description>IPv4 address to match</description>
+                  </valueHelp>
+                  <valueHelp>
+                    <format>ipv4net</format>
+                    <description>IPv4 prefix to match</description>
+                  </valueHelp>
+                  <valueHelp>
+                    <format>!ipv4</format>
```

Comment:
Did you test it?

```
set dont-throttle-netmasks 10.0.0.1
set dont-throttle-netmasks !10.0.0.2
set dont-throttle-netmasks !192.0.2.5
set dont-throttle-netmasks 10.0.0.2
```

Reply:
```
vyos@gw-01# edit service dns forwarding
[edit service dns forwarding]
vyos@gw-01# set dont-throttle-netmasks 10.0.0.1
[edit service dns forwarding]
vyos@gw-01# set dont-throttle-netmasks !10.0.0.2
[edit service dns forwarding]
vyos@gw-01# set dont-throttle-netmasks !192.0.2.5
[edit service dns forwarding]
vyos@gw-01# set dont-throttle-netmasks 10.0.0.2
[edit service dns forwarding]
vyos@gw-01# compare
[forwarding]
+ dont-throttle-netmasks "10.0.0.1"
+ dont-throttle-netmasks "!10.0.0.2"
+ dont-throttle-netmasks "!192.0.2.5"
+ dont-throttle-netmasks "10.0.0.2"

[edit service dns forwarding]
vyos@gw-01# commit;save
[edit]
vyos@gw-01# show service dns forwarding do
domain                  dont-throttle-netmasks
[edit]
vyos@gw-01# show service dns forwarding dont-throttle-netmasks
 dont-throttle-netmasks 10.0.0.1
 dont-throttle-netmasks !10.0.0.2
 dont-throttle-netmasks !192.0.2.5
 dont-throttle-netmasks 10.0.0.2
[edit]
vyos@gw-01#

vyos@gw-01# cat /run/powerdns/recursor.conf
### Autogenerated by service_dns_forwarding.py ###

# XXX: pdns recursor doesn't like whitespace near entry separators,
# especially in the semicolon-separated lists of name servers.
# Please be careful if you edit the template.

# Non-configurable defaults
daemon=yes
threads=1
allow-from=xxxx
log-common-errors=yes
non-local-bind=yes
query-local-address=0.0.0.0,::
lua-config-file=recursor.conf.lua

# cache-size
max-cache-entries=1000000

# negative TTL for NXDOMAIN
max-negative-ttl=0

# timeout
network-timeout=1500

# ignore-hosts-file
export-etc-hosts=yes

# listen-address
local-address=xxxx

# listen-port
local-port=53

# dnssec
dnssec=off


# dont-throttle-netmasks
dont-throttle-netmasks=10.0.0.1,!10.0.0.2,!192.0.2.5,10.0.0.2
# serve rfc1918 records
serve-rfc1918=yes

# zones
auth-zones=
forward-zones-file=recursor.forward-zones.conf[edit]

```

Reply:
what is sad is that the official docs say that negation is supported: https://docs.powerdns.com/recursor/settings.html#dont-throttle-netmasks

but digging into git for the version that vyos is using: https://github.com/PowerDNS/pdns/blob/566ad636d52331435705fb3a9c876966e53a4d9d/pdns/recursordist/docs/settings.rst
it doesn't say anything about negation.
I will remove the negation part.

Reply:
```
vyos@gw-01# sudo rec_control --socket-dir=/run/powerdns get-dont-throttle-netmasks
10.0.0.1/32, 10.0.0.2/32
```
after i removed the negated entries pdns responded :)

Cheers!